packages/ketcher-core/src/domain/entities/DrawingEntitiesManager.ts

```diff
@@ -175,6 +175,29 @@ export class DrawingEntitiesManager {
     return command;
   }
 
+  public selectCurrentDrawingEntities(drawingEntity: DrawingEntity) {
+    const command = new Command();
+
+    this.allEntities.forEach(([, drawingEntity]) => {
+      if (drawingEntity.selected) {
+        const operation = new DrawingEntitySelectOperation(drawingEntity);
+        command.addOperation(operation);
+      }
+    });
+
+    if (drawingEntity.selected) {
+      drawingEntity.turnOffSelection();
+      const operation = new DrawingEntitySelectOperation(drawingEntity);
+      command.addOperation(operation);
+    } else {
+      drawingEntity.turnOnSelection();
+      const operation = new DrawingEntitySelectOperation(drawingEntity);
+      command.addOperation(operation);
+    }
+
+    return command;
+  }
+
```

Comment:
I suggest below changes:

`  private addOperationToCommand(
    command: Command,
    drawingEntity: DrawingEntity,
  ) {
    const operation = new DrawingEntitySelectOperation(drawingEntity);
    command.addOperation(operation);
  }

  public selectCurrentDrawingEntities(drawingEntity: DrawingEntity) {
    const command = new Command();

    this.allEntities.forEach(([, drawingEntity]) => {
      if (drawingEntity.selected) {
        this.addOperationToCommand(command, drawingEntity);
      }
    });

    if (drawingEntity.selected) {
      drawingEntity.turnOffSelection();
    } else {
      drawingEntity.turnOnSelection();
    }
    this.addOperationToCommand(command, drawingEntity);

    return command;
  }`

Reply:
I believe that addOperationToCommand is not the best name for this method, because it actually adds concrete DrawingEntitySelectOperation. 
However I would propose to not create this method at all. We can do like this in selectCurrentDrawingEntities method:
```
      command.addOperation(
         new DrawingEntitySelectOperation(drawingEntity)
      );
```